src/AWS4AuthRequest.jl

```diff
@@ -5,30 +5,37 @@ using Dates
 using HTTP
 using HTTP.Pairs
 using HTTP.URIs
-using HTTP: Headers, Layer, request
+using HTTP: Headers, Layer, Request
 using IniFile
 using MbedTLS
 
 export AWS4AuthLayer
 
 """
-    request(AWS4AuthLayer, ::URI, ::Request, body) -> HTTP.Response
-
-Add a [AWS Signature Version 4](http://docs.aws.amazon.com/general/latest/gr/signature-version-4.html)
-`Authorization` header to a `Request`.
+    AWS4AuthLayer{Next} <: HTTP.Layer
 
+Abstract type used by [`HTTP.request`](@ref) to add an
+[AWS Signature v4](http://docs.aws.amazon.com/general/latest/gr/signature-version-4.html)
+authentication layer to the request.
+"""
+abstract type AWS4AuthLayer{Next<:Layer} <: Layer end
 
-Credentials are read from environment variables `AWS_ACCESS_KEY_ID`,
-`AWS_SECRET_ACCESS_KEY` and `AWS_SESSION_TOKEN`.
 """
-abstract type AWS4AuthLayer{Next <: Layer} <: Layer end
+    HTTP.request(AWS4AuthLayer, url::HTTP.URI, req::HTTP.Request, body) -> HTTP.Response
+
+Perform the given request, adding a layer of AWS authentication using
+[AWS Signature v4](http://docs.aws.amazon.com/general/latest/gr/signature-version-4.html).
+An "Authorization" header to the request.
 
-function HTTP.request(::Type{AWS4AuthLayer{Next}}, url::URI, req, body; kw...) where Next
+Credentials used for the signing are read from environment variables `AWS_ACCESS_KEY_ID`,
+`AWS_SECRET_ACCESS_KEY`, and `AWS_SESSION_TOKEN`.
```

Comment:
Credentials could also be read from a file or be instance based. May be best to just be less specific

Reply:
I've just deleted the mention of credentials here.